config.go

```diff
@@ -394,9 +408,32 @@ func (cfg *Config) unshadow() {
 	cfg.shadow = nil
 }
 
+func generateSwarmSecret() (string, error) {
+	key := make([]byte, 32)
+	_, err := crand.Read(key)
+	if err != nil {
+		return "", fmt.Errorf("Error reading from rand: %v", err)
+	}
+	return hex.EncodeToString(key), nil
+}
+
+func swarmSecretToKey(secret string) (string, error) {
+	hexSecret := hex.EncodeToString([]byte(secret))[:68]
+	if len(hexSecret) < 10 {
+		return "", fmt.Errorf("Swarm secret must be >= 10 chars, but is %d\n",
+			len(hexSecret))
+	}
+	var key bytes.Buffer
+	key.WriteString("/key/swarm/psk/1.0.0/\n")
+	key.WriteString("/base16/\n")
+	key.WriteString(hexSecret)
+
+	return key.String(), nil
+}
+
 // NewDefaultConfig returns a default configuration object with a randomly
 // generated ID and private key.
-func NewDefaultConfig() (*Config, error) {
+func NewDefaultConfig(swarmSecret *string) (*Config, error) {
```

Comment:
Regarding your comment on pointers, I've actually been curious about whether it's considered good form to use a pointer in place of what Haskell/Rust would refer to as `Nothing` or `None`, resp. E.g. if we want an empty string (`""`) to be distinguished from 'no input', it seems that the best option we have in Go is to use a pointer to a string and use `nil` to mean 'no input' (which I definitely think is bad practice (subverting the type system), but is sort of made to be the common solution based on the design of Go (and many languages)). Maybe there's a better alternative though?

Also, regarding this case in particular, is there any good reason in allowing an empty swarm secret/key (`""`)?